plugins/hydra_tsp_launcher/hydra_plugins/hydra_tsp_launcher/tsp_launcher.py

```diff
@@ -0,0 +1,222 @@
+        task_function: TaskFunction,
+    ) -> None:
+        self.config = config
+        self.config_loader = config_loader
+        self.task_function = task_function
+
+        hydra_overrides = []
+        for arg in sys.argv:
+            if '=' in arg and arg.startswith('hydra.'):
+                hydra_overrides.append(arg)
+
+        self.hydra_overrides = hydra_overrides
+        self.cmd_prefix = f"{sys.executable} {sys.argv[0]}"
+        self.tsp_prefix = look_for_task_spooler()
+        self.cmd_prefix = f"{self.tsp_prefix} {self.cmd_prefix}"
+
+    def tail_job(self, job_id):
+        job_status = _get_job_status(self.tsp_prefix, job_id)
+        while job_status == 'queued':
+            time.sleep(1)
+            job_status = _get_job_status(self.tsp_prefix, job_id)
+        
+        log.info(f"{job_id} has status {job_status}")
+        subprocess.call(
+            f"{self.tsp_prefix} -t {job_id}", shell=True
+        )
+
+    def launch(
+        self, job_overrides: Sequence[Sequence[str]], initial_job_idx: int
+    ) -> Sequence[JobReturn]:
+        """
+        :param job_overrides: a List of List<String>, where each inner list is the arguments for one job run.
+        :param initial_job_idx: Initial job idx in batch.
+        :return: an array of return values from run_job with indexes corresponding to the input list indexes.
+        """
+        setup_globals()
+        assert self.config is not None
+        assert self.config_loader is not None
+        assert self.task_function is not None
+
+        configure_log(self.config.hydra.hydra_logging, self.config.hydra.verbose)
+        sweep_dir = Path(str(self.config.hydra.sweep.dir))
+        sweep_dir.mkdir(parents=True, exist_ok=True)
+        log.info(
+            f"TaskSpooler Launcher is launching {len(job_overrides)} jobs locally"
+        )
+        log.info(f"Sweep output dir : {sweep_dir}")
+        runs = []
+        singleton_state = Singleton.get_state()        
+
+        for idx, overrides in enumerate(job_overrides):
+            overrides = list(overrides)
+            overrides.extend(self.hydra_overrides)
+            overrides = tuple(overrides)
+
+            ret = execute_job(
+                initial_job_idx + idx,
+                overrides,
+                self.config_loader,
+                self.config,
+                self.task_function,
+                singleton_state,
+                self.cmd_prefix,
+                self.tsp_prefix,
+            )
+            runs.append(ret)
+            time.sleep(self.time_between_submit)
+        
+        assert isinstance(runs, List)
+        for run in runs:
+            assert isinstance(run, JobReturn)
+
+        if self.tail_jobs:
+            Parallel(n_jobs=len(job_overrides), backend='threading')(
+                delayed(self.tail_job)(
+                    run.return_value
+                )
+                for run in runs
+            )
+        return runs
```

Comment:
Feels like a pretty big overkill to pull in a joblib dependency for this functionality.